### ISSUE TITLE
Mypy errors not recognised by travis

### ISSUE THREAD

**Issue by drvinceknight** [Axelrod-Python/Axelrod#868]
#867 and #839 show that travis is passing even if mypy fails. Here is some of the travis output:

```
axelrod/moran.py:335: error: Argument 1 to "fitness_proportionate_selection" has incompatible type "Tuple[str]"; expected Sequence[int]
axelrod/moran.py:342: error: Cannot determine type of 'players'
axelrod/moran.py:355: error: Need type annotation for variable
axelrod/moran.py:368: error: Cannot determine type of 'players'
axelrod/moran.py:374: error: Incompatible return value type (got Set[Tuple[int, int]], expected "Tuple[str]")
axelrod/moran.py:378: error: Cannot determine type of 'players'
axelrod/moran.py:380: error: Incompatible return value type (got Counter[str], expected "int")
The command "sh ./type_tests.sh" exited with 0.
```

Not sure what the fix is, but let's keep an eye on travis when merging @marcharper @meatballs (#839 was my fault for not checking this before giving a positive review label).

**Comment by drvinceknight** [comment]
I'm assuming the relevant part is:

```
The command "sh ./type_tests.sh" exited with 0.
```

**Comment by drvinceknight** [comment]
I wonder if this approach is worth looking at: https://github.com/zulip/zulip/pull/639/files

**Comment by drvinceknight** [comment]
As noted on #876 `Punisher` got merged with an error in the type checking. Would be good to get this setup so travis complains properly...

**Comment by drvinceknight** [comment]
Closes by #878

**Comment by drvinceknight** [comment]
Woops, closed the wrong thing!

**Comment by drvinceknight** [comment]
Closed by #878
